src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -78,6 +81,69 @@ public Storage storageForBucket(GoogleBucketResource bucketResource) {
     return gcsProjectFactory.getStorage(bucketResource.projectIdForBucket());
   }
 
+  /**
+   * Get all of the lines from any files matching the path, including wildcarded paths
+   *
+   * @param path path to files, or path including wildcard referring to many files
+   * @param projectId Project ID to use for storage service in case of requester pays bucket
+   * @return All of the lines from all of the files matching the path
+   */
+  public List<String> getGcsFilesLines(String path, String projectId) {
+    Storage storage = gcsProjectFactory.getStorage(projectId);
+    String prefixPath = path.contains("*") ? path.substring(0, path.lastIndexOf("*")) : path;
```

Comment:
It's not a big deal but this effectively calls `indexOf()` twice.
```suggestion
    int lastWildcard = path.substring(0, path.lastIndexOf("*"));
    String prefixPath = lastWildcard >= 0 ? path.substring(0, lastWildcard) : path;
```